spec/unit/pops/parser/parser_rspec_helper.rb

```diff
@@ -4,6 +4,17 @@
 
 module ParserRspecHelper
   include FactoryRspecHelper
+
+  def with_app_management(flag)
+    # Horrible things have to be done to test this as the Lexer sets up the
+    # KEYWORD table as a frozen constant, and this list depends on the
+    # Puppet[:app_management] setting. These 'before all' and 'after all'
+    # clauses forces Ruby to reload the Lexer
+    Puppet[:app_management] = true
```

Comment:
Change to `Puppet[:app_management] = flag` instead of always setting it to `true`.